executor/src/witgen/mod.rs

```diff
@@ -227,7 +228,13 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
             // as they are assumed to be handled in stage 0.
             let polynomial_identities = identities
```

Comment:
I'm curious what happens here if `identities` still contains anything else than polynomial identities? Due to the filtering they would get ignored? Or do we not have any non-polynomial identities, in which case this can be a hard cast rather than a filter?

Reply:
I think the filtering here could be improved in general. I don't see polynomial constraints between lower-stage witness column being filtered out, for example. I also don't see a reason why this should only contain polynomial constraints at this stage, e.g. we shouldn't filter out the lookups. Of course, usually when we use a multi-stage approach the idea is to remove the lookups, but conceptually, we could still have a lookup at a higher stage (although the backend probably would not support it).